sklearn/ensemble/tests/test_stacking.py

```diff
@@ -473,6 +480,75 @@ def test_stacking_cv_influence(stacker, X, y):
                         stacker_cv_5.final_estimator_.coef_)
 
 
+class NoFitRegressor(DummyRegressor):
+    def __init__(self, X, y, constant=0):
+        super().__init__(strategy="constant", constant=constant)
+        super().fit(X, y)
+
+    def fit(self, X, y, sample_weight=None):
+        raise AttributeError("fit is not allowed by this estimator!")
+
+
+class NoFitClassifier(DummyClassifier):
+    def __init__(self, X, y):
+        super().__init__(strategy="most_frequent")
+        super().fit(X, y)
+
+    def fit(self, X, y, sample_weight=None):
+        raise AttributeError("fit is not allowed by this estimator!")
+
+
+@pytest.mark.parametrize(
+    "stacker, X, y, acceptable_score",
+    [(StackingClassifier(
+        estimators=[
+            ('d0', NoFitClassifier(X=X_classification, y=y_classification)),
+            ('d1', NoFitClassifier(X=X_classification, y=y_classification))
+        ],
+        cv="prefit",
+        final_estimator=LogisticRegression()),
+      X_classification, y_classification, 0.8),
+     (StackingRegressor(
+         estimators=[
+             ('d0', NoFitRegressor(X=X_regression, y=y_regression)),
+             ('d1', NoFitRegressor(X=X_regression, y=y_regression, constant=1))
+         ],
+         cv="prefit",
+         final_estimator=LinearRegression()),
+      X_regression, y_regression, 0)],
+    ids=['StackingClassifier', 'StackingRegressor']
+)
+def test_stacking_prefit(stacker, X, y, acceptable_score):
+    # check if fit is not called when using prefit models
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, random_state=42
+    )
+    stacker.fit(X_train, y_train)
+    assert np.isclose(
+        stacker.score(X_test, y_test), acceptable_score, atol=0.05
+    )
```

Comment:
I think we can have a stronger test may checking that `fit` is not called again, that stack methods are called with the proper inputs, and `stacker.estimators_` are actually the same exact objects passed into `__init__`:

```python
from unittest.mock import Mock

@pytest.mark.parametrize(
    "Stacker, Estimator, stack_method, final_estimator, X, y",
    [(StackingClassifier, DummyClassifier, "predict_proba",
      LogisticRegression(random_state=42), X_iris, y_iris),
     (StackingRegressor, DummyRegressor, "predict",
      LinearRegression(), X_diabetes, y_diabetes)]
)
def test_stacking_prefit(Stacker, Estimator, stack_method, final_estimator, X, y):
    # check if fit is not called when using prefit models
    X_train1, X_train2, y_train1, y_train2 = train_test_split(
        X, y, random_state=42, test_size=0.5)
    estimators = [("d0", Estimator().fit(X_train1, y_train1)),
                  ("d1", Estimator().fit(X_train1, y_train1))]

	# mock out fit and stack_method to be asserted later
    for _, estimator in estimators:
        estimator.fit = Mock()
        stack_func = getattr(estimator, stack_method)
        setattr(estimator, stack_method, Mock(side_effect=stack_func))

    stacker = Stacker(
        estimators=estimators,
        cv="prefit",
        final_estimator=final_estimator
    )
    stacker.fit(X_train2, y_train2)

    assert stacker.estimators_ == [estimator for _, estimator in estimators]
    # fit was not called again
    assert all(estimator.fit.call_count == 0
               for estimator in stacker.estimators_)

	# stack method is called with the proper inputs
    for estimator in stacker.estimators_:
        stack_func_mock = getattr(estimator, stack_method)
        stack_func_mock.assert_called_with(X_train2)
```

This way we do not need to create `NoFitRegressor` or `NoFitClassifier` and the test is more self contained.